testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -832,6 +832,28 @@ public void testDeleteAllRowsInPartition()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testDeleteAllRowsUnpartitioned()
+    {
+        withTemporaryTable("delete_all_rows", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s (id INT) WITH (transactional = true)", tableName));
+
+            log.info("About to insert");
+            onTrino().executeQuery(format("INSERT INTO %s VALUES (1), (2), (3)", tableName));
+
+            log.info("About to delete one row");
+            onTrino().executeQuery(format("DELETE FROM %s WHERE id = 2", tableName));
+
+            log.info("About to verify");
+            verifySelectForTrinoAndHive("SELECT * FROM " + tableName, "TRUE", row(1), row(3));
+
+            log.info("About to delete all rows");
+            onTrino().executeQuery("DELETE FROM " + tableName);
```

Comment:
Was the bug triggered for `DELETE` without any predicate, or for a delete with a predicate too?

we should test both cases

Reply:
Good idea.  I added a test that differs in this way:

```
            // A predicate sufficient to fool statistics-based optimization
            onTrino().executeQuery(format("DELETE FROM %s WHERE id != 2" + tableName));
```